### ISSUE TITLE
Microsoft to have an official Kotlin Extension

### ISSUE THREAD

**Issue by RobFosterNYC** [microsoft/vscode#176843]
I would like Microsoft to have a working relationship with JetBrains official Kotlin Language,
So the language can work properly within VSCode, as well as have features like Linting and proper Code Formatting.
Too many extensions out there that are broken, or want people to jump through too many hoops just to get something simple working.

I would like to be able to have an official Microsoft Extension for Kotlin, like they do for Python, C#, C++ and others.
Sometimes people just want to open a simple IDE text editor like VSCode to do some quick coding instead of opening
and full fledged IDE , place ktlint in every single project to get something done. 

Can Microsoft PLEASE make an official extension for Kotlin that has Code Completion, Linting, Semantic Highlighting, Debugging, Go-To Definition, Signature Help, Hover, Formatting, Document Symbols, Find References and others.

Thank you  :)

**Comment by alexr00** [comment]
There are several existing Kotlin extensions on the extension marketplace. Maybe one of those would work for your needs.

**Comment by VSCodeTriageBot** [comment]
We try to keep VS Code lean and we think the functionality you're asking for is great for a VS Code extension. Maybe you can already find one that suits you in the [VS Code Marketplace](https://aka.ms/vscodemarketplace). Just in case, in a few simple steps you can get started [writing your own extension](https://aka.ms/vscodewritingextensions). See also our [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Happy Coding!

**Comment by RobFosterNYC** [comment]
I tried those extensions.  that's the first thing I did and they cause more problems than they should.
I even get tons of error messages when compiling.  THIS is the reason why I posted this post.  SMH
Microsoft currently has many extensions for languages and they run perfectly. Microsoft just does not have Kotlin.

All I am asking for is for Microsoft to do the same for Kotlin, just like they do for Python, C++, C# and other languages Microsoft has created extensions for.
